numpy/core/tests/test_deprecations.py

```diff
@@ -1230,3 +1230,23 @@ def test_deprecated_module(self):
     def test_deprecated_attr(self):
         finfo = np.finfo(float)
         self.assert_deprecated(lambda: getattr(finfo, "machar"))
+
+
+class TestQuantileInterpolationDeprecation(_DeprecationTestCase):
+    # Deprecated 2021-11-08, NumPy 1.22
+    @pytest.mark.parametrize("func",
+        [np.percentile, np.quantile, np.nanpercentile, np.nanquantile])
+    def test_deprecated(self, func):
+        self.assert_deprecated(
+            lambda: func([0., 1.], 0., interpolation="linear"))
+        self.assert_deprecated(
+            lambda: func([0., 1.], 0., interpolation="nearest"))
+
+    @pytest.mark.parametrize("func",
+            [np.percentile, np.quantile, np.nanpercentile, np.nanquantile])
+    def test_both_passed(self, func):
+        with warnings.catch_warnings():
+            # catch the warning, but make sure it does not raise:
```

Comment:
"does not" <- "does"?

Reply:
The warning is given before the error is thrown.  So we make sure the `DeprecationWarning` does _not_ raise, yes.  (In order to test the error that comes after the warning).

Reply:
Maybe "catch the DeprecationWarning  so that it does not raise".